lib/ff/ruby/server/sdk/api/cf_client.rb

```diff
@@ -1,91 +1,63 @@
-
+require 'singleton'
 require_relative "../../generated/lib/openapi_client"
 require_relative "../common/closeable"
 require_relative "inner_client"
 
 class CfClient < Closeable
-
-  # Static:
-  class << self
-
-    @@instance = CfClient.new
-
-    def instance
-
-      @@instance
-    end
-  end
-
-  # Static - End
-
-  def initialize(api_key = nil, config = nil, connector = nil)
-
-    if config == nil
-
-      @config = ConfigBuilder.new.build
-    else
-
-      @config = config
+  include Singleton
+
+  def init(api_key: nil, config: nil, connector: nil)
+    # Only initialize if @client is nil to avoid reinitialization
+    unless @client
+      @config = config || ConfigBuilder.new.build
+      @client = InnerClient.new(api_key, @config, connector)
+      @config.logger.debug "Client initialized with API key: #{api_key}"
```

Comment:
I think it would be convenient to always return `@client` from `#init`

Reply:
Yes, great spot @phoffer .  Thanks 👍